openapi-docs/docs/DefaultApi.md

```diff
@@ -0,0 +1,224 @@
+# AmbianicFunctionsCollection.DefaultApi
+
+All URIs are relative to *https://33743be6-3197-4dd4-8471-50b3640320c9.mock.pstmn.io*
+
+Method | HTTP request | Description
+------------- | ------------- | -------------
+[**deleteSubscription**](DefaultApi.md#deleteSubscription) | **DELETE** /subscription | Delete an Ambianic&#39;s user subscription
+[**getSubscriptionData**](DefaultApi.md#getSubscriptionData) | **GET** /subscription | Get a user&#39;s subscription data
+[**sendNotification**](DefaultApi.md#sendNotification) | **POST** /notification | Send an event detection notification
+[**subscribeUser**](DefaultApi.md#subscribeUser) | **POST** /subscribe | Subscribe Ambianic user to premium notification
```

Comment:
@vickywane for naming and semantic consistency, we should consider using

`POST /subscription`
as an endpoint to add a new subscription for a user.

instead of a separate endpoint name

`POST /subscribe`


That way we end up with two clean end points

`/subscription` that handles all CRUD operations (created, read, update, delete) for user subscriptions.

and 

`/notification` for triggering premium notifications.

Reply:
>  @vickywane this comment is still unresolved. What protection do we have against the malicious use of these public APIs?

By default, Netlify doesn't provide any means of protection for the deployed endpoints. However, I have created a follow-up pull request ( here ) to implement protection of the endpoints by ensuring that each request received is passed through a middleware that checks the origin of the request using a similar approach from [Middlyjs](https://github.com/middyjs/middy).

Reply:
Okay. 

Something to note is that each time a change is made in the route, it also has to be updated in the Ambianic PWA.